test/adapter/json_api/top_level_links_test.rb

```diff
@@ -0,0 +1,101 @@
+require 'test_helper'
+
+module ActiveModel
+  class Serializer
+    class Adapter
+      class JsonApi
+        class TopLevelLinksTest < Minitest::Test
+          URI = 'http://example.com'
+
+          def setup
+            ActionController::Base.cache_store.clear
+            @blog = Blog.new(id: 1,
+                             name: 'AMS Hints',
+                             writer: Author.new(id: 2, name: "Steve"),
+                             articles: [Post.new(id: 3, title: "AMS")])
+          end
+
+          def load_adapter(paginated_collection, options = {})
+            options = options.merge(adapter: :json_api)
```

Comment:
housekeeping `merge!` will create one less object and options will get GC'd anyhow